File: tests/fixtures/vulnerable/eval.py
from flask import request

def calc():
    # EXPECT-FLOW code-injection
    return eval(request.args["expr"])

def run():
    # EXPECT-FLOW code-injection
    exec(request.form["code"])

def compiled():
    # EXPECT-FLOW code-injection
    return compile(request.args["src"], "<s>", "exec")

def safe():
    return eval("1 + 1")                        # SAFE: constant

File: tests/fixtures/vulnerable/eval.php
<?php
function calc() {
  // EXPECT-FLOW code-injection
  return eval($_GET['expr']);
}
function checked() {
  // EXPECT-FLOW code-injection
  assert($_POST['cond']);
}
function safe() {
  return eval('return 1 + 1;');                 // SAFE: constant
}
